File: iosApp/iosApp/Resources/Colors.swift
import SwiftUI
import iosExport

class Colors {
	var primary: SwiftUI.Color { fatalError("Must override") }
	var secondary: SwiftUI.Color { fatalError("Must override") }
	var transparent: SwiftUI.Color { fatalError("Must override") }
}

class LightColors: Colors {
	override var primary: SwiftUI.Color { MultiplatformResource.colors.primary.getColor() }
	override var secondary: SwiftUI.Color { MultiplatformResource.colors.secondary.getColor() }
	override var transparent: SwiftUI.Color { MultiplatformResource.colors.transparent.getColor() }
}

class DarkColors: Colors {
	override var primary: SwiftUI.Color { MultiplatformResource.colors.primary.getColor() }
	override var secondary: SwiftUI.Color { MultiplatformResource.colors.secondary.getColor() }
	override var transparent: SwiftUI.Color { MultiplatformResource.colors.transparent.getColor() }
}

File: androidApp/ui/src/main/kotlin/com/widmeyertemplate/ui/theme/Colors.kt
package com.widmeyertemplate.ui.theme
import android.content.Context
import androidx.compose.ui.graphics.Color
import com.widmeyertemplate.resources.MultiplatformResource

sealed class Colors {
	abstract val primary: Color
	abstract val secondary: Color
	abstract val transparent: Color

    data class Light(
        private val context: Context,
		override val primary: Color = Color(MultiplatformResource.colors.primary.getColor(context)),
		override val secondary: Color = Color(MultiplatformResource.colors.secondary.getColor(context)),
		override val transparent: Color = Color(MultiplatformResource.colors.transparent.getColor(context)),
    ): Colors()

    data class Dark(
        private val context: Context, 
		override val primary: Color = Color(MultiplatformResource.colors.primary.getColor(context)),
		override val secondary: Color = Color(MultiplatformResource.colors.secondary.getColor(context)),
		override val transparent: Color = Color(MultiplatformResource.colors.transparent.getColor(context)),
    ): Colors()
}